File: Scala/src/test/scala/MyServiceTest.scala
import myservice.MyService
import org.scalatest.{Matchers, FlatSpec}
import sso.{Request, SSOToken, SingleSignOnRegistry}

class MyServiceTest extends FlatSpec with Matchers {

  "MyService" should "reject requests with invalid SSOTokens" in {
    val myService = new MyService
    val response = myService.handleRequest(Request("Foo", SSOToken("")))
    response.text should equal("hello Foo!")
  }
}

File: TypeScript/my_service.test.ts

import {MyService, Request, Response} from './my_service'
import {SingleSignOnRegistry} from "./single_sign_on";

describe('MyService', ()=>{

    it('rejects invalid SSOToken',() => {
        const service = new MyService(new SingleSignOnRegistry())
        const response = service.handle_request(new Request("Foo", null))
        expect(response.text).not.toEqual("Hello Foo!")
    })
})
